### PR TITLE
Search for known adapter library in the current working directory on Windows

### PR DESCRIPTION
**Pull request by PatKamin**
This is related to the task no. 1 from the issue #128.

This PR allows us to run the hello_world example with the null adapter library on Windows. Additionally, this PR makes it possible that some more adapters could be loaded in the future (with an env var) on Windows.

### CODE REVIEW

File: source/loader/ur_loader.cpp [hunk 1]
```diff
@@ -25,6 +25,8 @@ namespace loader
             {
                 platforms.emplace_back();
                 platforms.rbegin()->handle = handle;
+            } else {
+                return UR_RESULT_ERROR_MODULE_LINK_FAILURE;
```
Comment (pbalcer) [inline source/loader/ur_loader.cpp:29]: I don't think we want to completely fail init if one of the libraries doesn't load for some reason. Someone might just happen to have a completely different file with the path we expect.

Once we have logging infrastructure, we can print a warning/info about failing to load a library.

**Comment by pbalcer** [comment]
Can you please add running tests to the windows build job?

File: source/loader/ur_loader.cpp (hunk 1 above)
Comment (PatKamin) [reply source/loader/ur_loader.cpp]: Done. Removed this failure.

**Comment by PatKamin** [comment]
> Can you please add running tests to the windows build job?

Done. Had to call binaries directly as I couldn't make tests ran with ctest work properly with changed cwd. I think it's good enough as a temporary approach until the .dll loading mechanism will be changed.

File: test/python/basic.py [hunk 2]
```diff
@@ -12,7 +12,9 @@
 
 # not ideal, but this solution is simple and portable. Alternative is to set
 # PYTHONPATH or test on installed sources.
-sys.path.insert(1, '../include')
+from pathlib import Path
+ur_module_path = str(Path(os.path.realpath(__file__)).parents[2].joinpath("include"))
```
Comment (pbalcer) [inline test/python/basic.py:16]: This is too hacky :-) Maybe instead we can just remove this entirely and set PYTHONPATH like this in the CMake:
```
set_property(TEST python-basic APPEND PROPERTY
                ENVIRONMENT "PYTHONPATH=${PROJECT_SOURCE_DIR}")
```
?

this should allow us to just `import include.ur as ur` instead.

File: include/ur.py [hunk 3]
```diff
@@ -1945,7 +1946,9 @@ class UR_DDI:
     def __init__(self, version : ur_api_version_t):
         # load the ur_loader library
         if "Windows" == platform.uname()[0]:
-            self.__dll = WinDLL("ur_loader.dll")
+            os.add_dll_directory(os.getcwd())
+            LOAD_LIBRARY_SEARCH_USER_DIRS = 0x00000400
+            self.__dll = WinDLL("ur_loader.dll", winmode=LOAD_LIBRARY_SEARCH_USER_DIRS)
```
Comment (pbalcer) [inline include/ur.py:1951]: Please check if `LOAD_LIBRARY_SEARCH_DEFAULT_DIRS` solves this automatically.

File: source/loader/windows/platform_discovery_win.cpp [hunk 4]
```diff
@@ -20,9 +21,40 @@
...
+    if (nullptr == result) {
+        return enabledPlatforms;
+    }
+    result = AddDllDirectory(libsPath);
+    if (nullptr == result) {
+        return enabledPlatforms;
+    }
+
+    // UR_ENABLE_ALT_DRIVERS is for development/debug only
+    altPlatforms = getenv("UR_ENABLE_ALT_DRIVERS");
```
[the remark this answers is not in the document]
Comment (PatKamin) [reply source/loader/windows/platform_discovery_win.cpp]: Opened #145

**Review by pbalcer — APPROVED** [review]

**Comment by pbalcer** [comment]
I still don't like having to create a windows-only env variable just so python tests work, but I can't think of a better idea for now.

**Pull request merged**